client/components/gsuite/gsuite-price/index.jsx

```diff
@@ -24,8 +24,8 @@ export const hasDiscount = ( product ) => {
 	}
 
 	const currentTime = Date.now();
-	const startDate = new Date( product.sale_coupon.start_date );
-	const endDate = new Date( product.sale_coupon.expires );
+	const startDate = new Date( product.sale_coupon.start_date + ' GMT' );
+	const endDate = new Date( product.sale_coupon.expires + ' GMT' );
```

Comment:
What about using `UTC` instead of `GMT`?

Reply:
I started testing this, and researched [the Date constructor on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date), and I realised that the date parsing might be browser-dependent. And I found that Firefox doesn't like `+ ' GMT'` or `+ ' UTC'`!

So I have reimplemented the code to handle this by converting to an ISO 8601 datetime with the `Z` suffix for UTC.

Reply:
I also had a look to that documentation last week but completely missed it wouldn't work with certain browsers. I'm glad you noticed that!